service/dap/server.go

```diff
@@ -1947,7 +1947,7 @@ func (s *Server) getTypeIfSupported(v *proc.Variable) string {
 	if !s.clientCapabilities.supportsVariableType {
 		return ""
 	}
-	return v.TypeString()
+	return api.ConvertVar(v).TypeString()
```

Comment:
`ConvertVar` does a bunch of extra things, I think you should just do, in service/dap:

```
if v.Kind == reflect.Interface && len(v.Children) > 0 {
    return fmt.Sprintf("%s(%s)", v.TypeString(), v.Children[0].TypeString())
}
return v.TypeString()
```

Note that you have to check that there are children, which you didn't in your method. Also I'm not sure about the reflect.Ptr case. Why just pointers? What about slices, arrays and maps?

Reply:
Pointers point to a single address, whereas slices, arrays, and maps may have different underlying types for each interface, so looking for the concrete type does not make sense.